In [1]:
import pandas as pd

# Load the file directly
check_df = pd.read_parquet('../slurm_files/LLM_DEGRADATION_RESULTS.parquet')

# 1. See the shape
print(f"Total Rows: {check_df.shape[0]}")
print(f"Total Columns: {check_df.shape[1]}")

# 2. The 'Truth Table' (The most important check)
print("\n--- EXPERIMENT CENSUS (City vs. Variant) ---")
census = pd.crosstab(check_df['city'], check_df['variant_type'])
print(census)

# 3. Look at the column names
print("\n--- AVAILABLE COLUMNS ---")
print(check_df.columns.tolist())

Total Rows: 22173
Total Columns: 7

--- EXPERIMENT CENSUS (City vs. Variant) ---
variant_type  mask_both  mask_directions  mask_landmark
city                                                   
manhattan          5207             6823           6485
philadelphia          0             1262              0
pittsburgh          632              984            780

--- AVAILABLE COLUMNS ---
['sample_id', 'city', 'variant_type', 'masked_instruction', 'original_text', 'gold_goal_node', 'llm_output_raw']


The output showcases a problem:

philadelphia          0             1262              0

This indicates that for the city of Philadelphia, there are 1262 entries for the 'degradation' variant type, but none for the 'control' variant type. This imbalance in the data could lead to biased results in any analysis or modeling efforts, as it does not provide a comparative baseline for the 'control' group in Philadelphia. It is crucial to investigate why there is a lack of 'control' entries for Philadelphia and consider how this might affect the overall conclusions drawn from the dataset.

Hence: `extraction_forensics.ipynb`